### PR TITLE
Feature/dynamic height for selects

### PR DESCRIPTION
**Pull request by zenangst**
**Before**
screen shot 2015-05-13 at 10 52 43

**After**
screen shot 2015-05-13 at 10 53 06

### CODE REVIEW

File: Source/Cells/Popover/FORMSelectFieldCell.m [hunk 1]
```diff
@@ -68,6 +69,18 @@ - (void)updateWithField:(FORMField *)field {
 - (void)updateContentViewController:(UIViewController *)contentViewController
                           withField:(FORMField *)field {
     self.fieldValuesController.field = self.field;
+
+    if (self.field.values.count <= FORMSelectMaxItemCount) {
+        CGSize currentSize = FORMSelectPopoverSize;
+        CGFloat headerViewHeight = self.fieldValuesController.headerView.frame.size.height;
+        CGFloat labelHeight = round(self.fieldValuesController.headerView.labelHeight);
+        CGSize customSize = {
+            .width = currentSize.width,
```
Comment (3lvis) [inline Source/Cells/Popover/FORMSelectFieldCell.m]: Can you use the normal CGSizeMake?

File: Source/Cells/Popover/FORMSelectFieldCell.m [hunk 2]
```diff
@@ -68,6 +69,18 @@ - (void)updateWithField:(FORMField *)field {
...
+            .height = (FORMFieldValuesCellHeight * self.field.values.count) + labelHeight + headerViewHeight + FORMTitleLabelY
+        };
+
+        [self.fieldValuesController setPreferredContentSize:customSize];
```
Comment (3lvis) [inline Source/Cells/Popover/FORMSelectFieldCell.m]: Do you need the setPreferredContentSize? Or you could do `self.fieldValuesController.preferredContentSize`

File: Source/Cells/Popover/FORMSelectFieldCell.m (hunk 1 above)
Comment (zenangst) [inline Source/Cells/Popover/FORMSelectFieldCell.m]: @3lvis Fixed https://github.com/hyperoslo/Form/commit/5ea997dbdbddb010217420779cf0b420b256a3db

File: Source/Cells/Popover/FORMSelectFieldCell.m [hunk 3]
```diff
@@ -68,6 +69,15 @@ - (void)updateWithField:(FORMField *)field {
...
                           withField:(FORMField *)field {
     self.fieldValuesController.field = self.field;
+
+    if (self.field.values.count <= FORMSelectMaxItemCount) {
+        CGSize currentSize = FORMSelectPopoverSize;
+        CGFloat headerViewHeight = self.fieldValuesController.headerView.frame.size.height;
+        CGFloat labelHeight = round(self.fieldValuesController.headerView.labelHeight);
+        CGSize customSize = CGSizeMake(currentSize.width, (FORMFieldValuesCellHeight * self.field.values.count) + labelHeight + headerViewHeight + FORMTitleLabelY);
+
+        [self.fieldValuesController setPreferredContentSize:customSize];
```
Comment (3lvis) [inline Source/Cells/Popover/FORMSelectFieldCell.m]: Do you need the setPreferredContentSize? Or you could do `self.fieldValuesController.preferredContentSize`

**Comment by 3lvis** [comment]
@zenangst Looks fantastic!

File: Source/Cells/Popover/FORMSelectFieldCell.m (hunk 3 above)
Comment (zenangst) [inline Source/Cells/Popover/FORMSelectFieldCell.m]: Done!

File: Source/Cells/Popover/FORMSelectFieldCell.m [hunk 4]
```diff
@@ -68,6 +69,15 @@ - (void)updateWithField:(FORMField *)field {
 - (void)updateContentViewController:(UIViewController *)contentViewController
                           withField:(FORMField *)field {
     self.fieldValuesController.field = self.field;
+
+    if (self.field.values.count <= FORMSelectMaxItemCount) {
+        CGSize currentSize = FORMSelectPopoverSize;
+        CGFloat headerViewHeight = self.fieldValuesController.headerView.frame.size.height;
+        CGFloat labelHeight = round(self.fieldValuesController.headerView.labelHeight);
```
Comment (3lvis) [inline Source/Cells/Popover/FORMSelectFieldCell.m]: You could move this to a separate method that you could test.

Comment (zenangst) [inline Source/Cells/Popover/FORMSelectFieldCell.m]: That would be a great addition, I can make an issue for that and fix it later on.

Comment (3lvis) [inline Source/Cells/Popover/FORMSelectFieldCell.m]: Perfect :)

**Pull request merged**